packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectHelper.kt

```diff
@@ -211,7 +234,81 @@ internal object RealmObjectHelper {
         val newValue = if (value?.`$realm$IsManaged` == false) {
             copyToRealm(obj.`$realm$Mediator`!!, obj.`$realm$Owner`!!, value)
         } else value
-        setValueByKey(obj, obj.propertyKeyOrThrow(propertyName), newValue)
+        setValueByKey(obj, obj.propertyInfoOrThrow(propertyName).key, newValue)
+    }
+
+    /**
+     * Get values for non-collection properties by name.
+     *
+     * This will verify that the requested type (`clazz`) and nullability matches the property
+     * properties in the schema.
+     */
+    internal fun <R : Any> dynamicGet(
+        obj: RealmObjectInternal,
+        propertyName: String,
+        clazz: KClass<R>,
+        nullable: Boolean
+    ): R? {
+        obj.checkValid()
+        val propertyInfo = checkPropertyType(obj, propertyName, CollectionType.RLM_COLLECTION_TYPE_NONE, clazz, nullable)
```

Comment:
I suspect we eventually want to move all of this to JNI so we don't have multiple JNI calls here, but "future improvement" (tm).

Reply:
Unless we do a `ClassMetadata` implementation that is backed by JNI calls then this should stay in Kotlin. The current implementation just uses the cached schema information from the schema.

Reply:
Good point. Then it probably doesn't matter as much 👍